analysis/src/github.rs

```diff
@@ -0,0 +1,154 @@
+//! This module abstracts the communication with GitHub API for a given crate
+
+use anyhow::{anyhow, Result};
+use git_url_parse::GitUrl;
+use guppy::graph::PackageMetadata;
+use reqwest::header::{HeaderMap, HeaderValue, AUTHORIZATION, USER_AGENT};
+use serde::{Deserialize, Serialize};
+
+pub struct GitHubReport {
+    pub name: String,       // name of the crate
+    pub repository: String, // repository url
+    pub is_github_repo: bool,
+    pub repo_stats: RepoStats,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+pub struct RepoStats {
+    pub full_name: Option<String>,
+    pub stargazers_count: u64,
+    pub subscribers_count: u64,
+}
+
+pub struct GitHubAnalyzer {
+    client: reqwest::blocking::Client,
+}
+
+impl GitHubAnalyzer {
+    fn construct_headers() -> HeaderMap {
+        let mut headers = HeaderMap::new();
+        headers.insert(USER_AGENT, HeaderValue::from_static("diem/whackadep"));
+
+        let pat =
+            std::env::var("GITHUB_TOKEN").expect("GITHUB_TOKEN is not set in the environment");
+        let pat = format!("token {}", pat);
+        let mut auth_value = HeaderValue::from_str(&pat).expect("Cannot parse token");
+        auth_value.set_sensitive(true);
+        headers.insert(AUTHORIZATION, auth_value);
+
+        headers
+    }
+
+    fn default(name: String, repository: String) -> GitHubReport {
+        //Returns a default GitHubReport with is_github_repo set as false
+        GitHubReport {
+            name,
+            repository,
+            is_github_repo: false,
+            repo_stats: RepoStats {
+                full_name: None,
+                ..Default::default()
+            },
+        }
+    }
```

Comment:
You should implement a constructor for the GitHubReport and have it be in its own impl block